profile/label_nodes.go

```diff
@@ -0,0 +1,97 @@
+package profile
+
```

Comment:
Could we move this file under internal/ directory? We limit the API surface of the pprof package and mainly expose the tool itself to the users. So I'd prefer not to expose any new public API in this PR.

Reply:
Yeah good call about the public API surface. I'll have a go at this. I'll have to copy over a lot of the test profile code because it'll be in a different module now, but maybe a little copying is OK.